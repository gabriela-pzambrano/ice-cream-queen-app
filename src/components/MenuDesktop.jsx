import React from 'react';
import logo from '../assets/logo-ice-cream-queen.svg';
import bottomIceCream from '../assets/bottom-ice-cream.svg';

function classNames(...classes) {
  return classes.filter(Boolean).join(' ');
}

const MenuDesktop = ({sidebarNavigation}) => {
  return (
    <div className="hidden w-32 overflow-y-auto bg-primary-700 md:block">
      <div className="flex w-full flex-col items-center py-6">
        <div className="flex flex-shrink-0 items-center">
          <img className="h-12 w-auto" src={logo} alt="Your Company" />
        </div>
        <div className="mt-6 w-full flex-1 space-y-1 px-2">
          {sidebarNavigation.map((item) => (
            <a
              key={item.name}
              href={item.href}
              className={classNames(
                item.current
                  ? 'bg-primary-800 text-white'
                  : 'text-primary-100 hover:bg-primary-800 hover:text-white',
                'group w-full p-3 rounded-md flex flex-col items-center text-xs font-medium'
              )}
              aria-current={item.current ? 'page' : undefined}
            >
              <item.icon
                className={classNames(
                  item.current
                    ? 'text-white'
                    : 'text-primary-300 group-hover:text-white',
                  'h-6 w-6'
                )}
                aria-hidden="true"
              />
              <span className="mt-2">{item.name}</span>
            </a>
          ))}
        </div>
      </div>
      <img
        className="w-32 absolute bottom-0"
        src={bottomIceCream}
        alt="Your Company"
      />
    </div>
  );
};

export default MenuDesktop;
