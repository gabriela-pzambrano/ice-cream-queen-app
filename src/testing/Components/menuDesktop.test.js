import { render, screen } from '@testing-library/react';
import MenuDesktop from '../../components/MenuDesktop';
/* import userEvent from '@testing-library/user-event'; */

jest.mock('../../assets/logo-ice-cream-queen.svg', () => 'ruta/de/la/imagen.jpg');
jest.mock('../../assets/bottom-ice-cream.svg', () => 'ruta/de/la/imagen.jpg');

describe('MenuDesktop', () => {

  test('MenuDesktop should render the correct navigation links', () => {
    const sidebarNavigation = [
      { name: 'Home', icon: () => null },
      { name: 'About', icon: () => null },
      { name: 'Contact', icon: () => null },
    ];

    const selectedTab = { name: 'Home', current: true };
    const setSelectedTab = jest.fn();

    render(<MenuDesktop
      sidebarNavigation={sidebarNavigation}
      selectedTab={selectedTab}
      setSelectedTab={setSelectedTab} />);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('About')).toBeInTheDocument();
    expect(screen.getByText('Contact')).toBeInTheDocument();
  });

  /* test('activates selected tab', () => {

    const sidebarNavigation = [
      { name: 'Home', icon: () => 'ruta/de/la/imagen.jpg' },
      { name: 'About', icon: () => 'ruta/de/la/imagen.jpg'},
      { name: 'Contact', icon: () => 'ruta/de/la/imagen.jpg' },
    ];

    const selectedTab = { name: 'Home' };
    const setSelectedTab = jest.fn();
  
    render(
      <MenuDesktop
        sidebarNavigation={sidebarNavigation}
        selectedTab={selectedTab}
        setSelectedTab={setSelectedTab}
      />
    );
  
    const aboutTabElement = screen.getByText('About');
    userEvent.click(aboutTabElement);
  
    expect(setSelectedTab).toHaveBeenCalled();
    expect(setSelectedTab).toHaveBeenCalledWith({ name: 'About', icon: () => 'ruta/de/la/imagen.jpg' , current: true });
  }); */

  test('renders logo', () => {
    const sidebarNavigation = [
      { name: 'Home', icon: () => <div></div> },
      { name: 'About', icon: () => <div></div> },
      { name: 'Contact', icon: () => <div></div> },
    ];
    const selectedTab = { name: 'Home' };
    const setSelectedTab = () => {};
  
    render(
      <MenuDesktop
        sidebarNavigation={sidebarNavigation}
        selectedTab={selectedTab}
        setSelectedTab={setSelectedTab}
      />
    );
  
    const logoElement = screen.getByAltText('ice-cream-queen');
    expect(logoElement).toBeInTheDocument();
  });

  test('renders footer image', () => {
    const sidebarNavigation = [
      { name: 'Home', icon: () => <div></div> },
      { name: 'About', icon: () => <div></div> },
      { name: 'Contact', icon: () => <div></div> },
    ];
    const selectedTab = { name: 'Home' };
    const setSelectedTab = () => {};
  
    render(
      <MenuDesktop
        sidebarNavigation={sidebarNavigation}
        selectedTab={selectedTab}
        setSelectedTab={setSelectedTab}
      />
    );
  
    const footerImageElement = screen.getByAltText('Ice Cream Queen');
    expect(footerImageElement).toBeInTheDocument();
  });



});
