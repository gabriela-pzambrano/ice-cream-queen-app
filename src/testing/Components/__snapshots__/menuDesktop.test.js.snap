// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDesktop Check that MenuDesktop renders correctly 1`] = `
<div>
  <div
    class="hidden w-36 overflow-y-auto bg-primary-700 md:block"
  >
    <div
      class="flex w-full flex-col items-center py-6"
    >
      <div
        class="flex flex-shrink-0 items-center"
      >
        <img
          alt="Your Company"
          class="h-12 w-auto"
          src="ruta/de/la/imagen.jpg"
        />
      </div>
      <div
        class="mt-6 w-full flex-1 space-y-1 px-2"
      >
        <a
          class="text-primary-100 hover:bg-primary-800 hover:text-white group w-full p-3 rounded-md flex flex-col items-center text-xs font-medium"
          href="/"
        >
          <span
            class="mt-2"
          >
            Home
          </span>
        </a>
        <a
          class="text-primary-100 hover:bg-primary-800 hover:text-white group w-full p-3 rounded-md flex flex-col items-center text-xs font-medium"
          href="/about"
        >
          <span
            class="mt-2"
          >
            About
          </span>
        </a>
      </div>
    </div>
    <img
      alt="Your Company"
      class="w-36 absolute bottom-0"
      src="ruta/de/la/imagen.jpg"
    />
  </div>
</div>
`;
